nmi_mysql/nmi_mysql.py

```diff
@@ -11,7 +11,31 @@
 
 class DB(object):
 
-    def __init__(self, conf, autoconnect=False):
+    def __init__(self, conf, max_pool_size=20):
+        self.conf = conf
+        self.max_pool_size = max_pool_size
+        self._initialize_pool()
+
+    def _initialize_pool(self):
+        self.pool = Queue(maxsize=self.max_pool_size)
+
+        for _ in range(0, self.max_pool_size):
+            self.pool.put_nowait(Connection(self.conf))
+
+    def connect(self):
+        con = self.pool.get(True)
+        con.connect()
+
+        return con
+
+    def close(self, con):
+        con.close()
+        self.pool.put_nowait(con)
```

Comment:
I like where this is going, but the connection should not be closed, the point of pooling is that the connection is being re-used so there's no need to create a new connection. If by close, you meant by release back to the pool (so that there's no need to login/create the connection again), then that's fine.